c-scape/src/mem/ntbs.rs

```diff
@@ -111,6 +111,134 @@ unsafe extern "C" fn strcmp(mut s1: *const c_char, mut s2: *const c_char) -> c_i
     *s1 as c_uchar as c_int - *s2 as c_uchar as c_int
 }
 
+// enum for strverscmp state
+// internal so no surface
+// see https://codebrowser.dev/glibc/glibc/string/strverscmp.c.html#26
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+enum StrverscmpState {
+    Normal,
+    Integral,
+    Fractional,
+    LeadingZeros,
+}
+
+enum CharType {
+    Zero,
+    Digit,
+    NonNumeric,
+}
+
+impl CharType {
+    fn from_char(c: c_char) -> Self {
+        match c {
+            // ASCII 0
+            48 => Self::Zero,
```

Comment:
Instead of hard-coding 48, could you (cast `c` to `u8` and) write this as b`0`, and the following case as `b'1'..=b'9'`?